service-discovery/combined/src/main/java/io/smallrye/stork/servicediscovery/combined/CombinedServiceDiscovery.java

```diff
@@ -0,0 +1,40 @@
+package io.smallrye.stork.servicediscovery.combined;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.groups.UniJoin;
+import io.smallrye.stork.ServiceDiscovery;
+import io.smallrye.stork.ServiceInstance;
+import io.smallrye.stork.Stork;
+import io.smallrye.stork.servicediscovery.combined.util.CombiningList;
+
+public class CombinedServiceDiscovery implements ServiceDiscovery {
+
+    private final Collection<String> serviceNames;
+    private final List<ServiceDiscovery> elements = new ArrayList<>();
+
+    public CombinedServiceDiscovery(Collection<String> serviceNames) {
+        this.serviceNames = Collections.unmodifiableCollection(serviceNames);
+    }
+
+    @Override
+    public Uni<List<ServiceInstance>> getServiceInstances() {
+        UniJoin.Builder<List<ServiceInstance>> builder = Uni.join().builder();
+        for (ServiceDiscovery element : elements) {
+            builder.add(element.getServiceInstances());
+        }
+
+        return builder.joinAll().andFailFast().onItem().transform(CombiningList::new);
```

Comment:
I'm not sure about the failFast. Maybe that should be a configuration.
We may have the case where you use "combined services" to cover multiple environments. When you run, some service discovery may not be accessible and so fails.

Reply:
There's nothing built in for this, right? The other strategy also returns failure on an error but combines all errors.

Reply:
I reworked the other things, I'd rather add support for ignoring a failure later, WDYT @cescoffier ?

Reply:
+1